### ISSUE TITLE
Formatting sometimes loses closing #endregion directive when transforming to file-scoped namespace

### ISSUE THREAD

**Issue by madelson** [dotnet/format#1578]
This is with a C# project with LangVersion 10 using dotnet-format 7.0.321101. 

I'm running `dotnet-format My.sln --diagnostics=IDE0161`.

As a note, I saw this issue a lot with `dotnet format` in the .NET 6 SDK. With the latest version of `dotnet-format` from this repo it usually doesn't happen but still does in a few cases.

Example files (weird naming/comments is due my stripping out any business-related content while trying to avoid structural changes to the file):

.editorconfig
```
[*.{cs,cshtml,css,js,jsx,less,ts,tsx}]
indent_style = tab
[*.cs]
# Default to file-scoped namespaces
csharp_style_namespace_declarations = file_scoped:warning
```

**BEFORE**
```cs
using System;
using System.Text.RegularExpressions;

namespace N
{
	/// <summary>
	/// c
	/// </summary>
	public sealed class C
	{
		// forbid external construction
		internal C()
		{
		}

		internal RegexOptions? R { get; set; }

		// c
		internal int S { get; set; }

		// c
		internal int? E { get; set; }

		/// <summary>
		/// c
		/// </summary>
		internal TimeSpan? Cadence { get; set; }

		/// <summary>
		/// c
		/// </summary>
		/// <param name="i">i</param>
		public B F(int i)
		{
			this.V(i, "i");
			this.S = i;

			this.TrySet(RegexOptions.Compiled);

			return new B(this);
		}

		/// <summary>
		/// c
		/// </summary>
		/// <param name="t">t</param>
		public Ccb Every(TimeSpan t)
		{
			this.TrySet(RegexOptions.Singleline);

			this.Cadence = t;
			return new Ccb(this);
		}

		internal static object Create(Action<C> a)
		{
			var builder = new C();
			a(builder);

			return new object();
		}

		internal void V(int h, string s)
		{
		}

		private void TrySet(RegexOptions o)
		{
			this.R = o;
		}
	}

	#region --- R ---

	/// <summary>
	/// C
	/// </summary>
	public sealed class B
	{
		internal B(C c)
		{
			this._c = c;
		}

		/// <summary>
		/// c
		/// </summary>
		private readonly C _c;

		/// <summary>
		/// c
		/// </summary>
		/// <param name="i">i</param>
		public C ThroughHour(int i)
		{
			this._c.V(i, "i");
			this._c.E = i;

			return this._c;
		}
	}

	/// <summary>
	/// c
	/// </summary>
	public sealed class Ccb
	{
		internal Ccb(C c)
		{
			this._c = c;
		}

		/// <summary>
		/// c
		/// </summary>
		private readonly C _c;

		/// <summary>
		/// c
		/// </summary>
		/// <param name="i">i</param>
		public Cchrb FromHour(int i)
		{
			this._c.V(i, "i");
			this._c.S = i;

			return new Cchrb(this._c);
		}
	}

	/// <summary>
	/// c
	/// </summary>
	public sealed class Cchrb
	{
		internal Cchrb(C c)
		{
			this._c = c;
		}

		/// <summary>
		/// c
		/// </summary>
		private readonly C _c;

		/// <summary>
		/// c
		/// </summary>
		/// <param name="i">i</param>
		public Cchrb T(int i)
		{
			this._c.V(i, "i");
			this._c.E = i;

			return this;
		}
	}

	#endregion
}
```

**AFTER**
```cs
using System;
using System.Text.RegularExpressions;

namespace N;

/// <summary>
/// c
/// </summary>
public sealed class C
{
	// forbid external construction
	internal C()
	{
	}

	internal RegexOptions? R { get; set; }

	// c
	internal int S { get; set; }

	// c
	internal int? E { get; set; }

	/// <summary>
	/// c
	/// </summary>
	internal TimeSpan? Cadence { get; set; }

	/// <summary>
	/// c
	/// </summary>
	/// <param name="i">i</param>
	public B F(int i)
	{
		this.V(i, "i");
		this.S = i;

		this.TrySet(RegexOptions.Compiled);

		return new B(this);
	}

	/// <summary>
	/// c
	/// </summary>
	/// <param name="t">t</param>
	public Ccb Every(TimeSpan t)
	{
		this.TrySet(RegexOptions.Singleline);

		this.Cadence = t;
		return new Ccb(this);
	}

	internal static object Create(Action<C> a)
	{
		var builder = new C();
		a(builder);

		return new object();
	}

	internal void V(int h, string s)
	{
	}

	private void TrySet(RegexOptions o)
	{
		this.R = o;
	}
}

#region --- R ---

/// <summary>
/// C
/// </summary>
public sealed class B
{
	internal B(C c)
	{
		this._c = c;
	}

	/// <summary>
	/// c
	/// </summary>
	private readonly C _c;

	/// <summary>
	/// c
	/// </summary>
	/// <param name="i">i</param>
	public C ThroughHour(int i)
	{
		this._c.V(i, "i");
		this._c.E = i;

		return this._c;
	}
}

/// <summary>
/// c
/// </summary>
public sealed class Ccb
{
	internal Ccb(C c)
	{
		this._c = c;
	}

	/// <summary>
	/// c
	/// </summary>
	private readonly C _c;

	/// <summary>
	/// c
	/// </summary>
	/// <param name="i">i</param>
	public Cchrb FromHour(int i)
	{
		this._c.V(i, "i");
		this._c.S = i;

		return new Cchrb(this._c);
	}
}

/// <summary>
/// c
/// </summary>
public sealed class Cchrb
{
	internal Cchrb(C c)
	{
		this._c = c;
	}

	/// <summary>
	/// c
	/// </summary>
	private readonly C _c;

	/// <summary>
	/// c
	/// </summary>
	/// <param name="i">i</param>
	public Cchrb T(int i)
	{
		this._c.V(i, "i");
		this._c.E = i;

		return this;
	}
}
```

**Comment by JoeRobich** [comment]
@madelson Thanks for reporting this issue. The removal of pragmas in trailing trivia was also reported in https://github.com/dotnet/roslyn/issues/58510. The issue was resolved in https://github.com/dotnet/roslyn/pull/58434. A fix will ship in the 6.0.4xx SDK.

**Comment by madelson** [comment]
@JoeRobich I just want to confirm that this has been fixed as of **dotnet-format 7.0.321101**.

I had noticed that this problem was very common in `dotnet format` with the current 6.x SDK which is why I installed the newer version. With 7.0.321101 I saw this happening much less frequently but it still repro'd with the file I posted.
